src/UI/Conversation.jsx

```diff
@@ -146,33 +155,58 @@ class Conversation extends Component {
 				<div className={styles.body}>
 					{
 						this.state.welcomeMessage
-							&& <Announcement message={this.state.welcomeMessage} />
+						&& <Announcement message={this.state.welcomeMessage} />
 					}
 					{
 						this.state.messages.map((message, index, array) => (
 							<React.Fragment key={message.id}>
 								{
 									this.setRenderedDate(this.getDateFromTimestamp(message.time))
-										&& <DateGroup timestamp={message.time} />
+									&& <DateGroup timestamp={message.time} />
+								}
+								{
+									message.carousel.length === 0
+									&& <Message
+										api={this.props.api}
+										message={message}
+										showAgent={this.shouldRenderAgentName(index, index - 1)}
+									   />
+								}
+								{
+									message.carousel && message.carousel.length > 0
```

Comment:
Hier kijk je wel of carousel er is maar kun je korter schrijven door `message.carousel?.length` te doen.

Reply:
Hmm en kunnen we niet zeggen dat als deze false is dan altijd een message is, dus als fallback van carousel gewoon message is, if/else